### PR TITLE
chore(sdk): fix error 'ApiResourceReference' object has no attribute 'id'

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
Fixes the bug in https://github.com/kubeflow/pipelines/pull/6934#issuecomment-984412593.
PR title is chore instead of "fix", because the bug is fixed before a release.

I verified these fixes locally, they resolve the error message.

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by Bobgy** [comment]
/cc @alexlatchford 
I think we also need to cherry pick to 1.8 branch.

File: sdk/python/kfp/cli/run.py [hunk 1]
```diff
@@ -222,7 +222,7 @@ def _print_runs(runs: List[kfp_server_api.ApiRun], output_format: OutputFormat):
         run.created_at.isoformat(),
         next(rr
              for rr in run.resource_references
-             if rr.key.type == kfp_server_api.ApiResourceType.EXPERIMENT).id
+             if rr.key.type == kfp_server_api.ApiResourceType.EXPERIMENT).key.id
```
Comment (alexlatchford) [inline sdk/python/kfp/cli/run.py:225]: Hey @Bobgy thanks for cleaning this up and apologies about the error this was a last minute change to make it a little nicer. Likely I've made the same mistake [here](https://github.com/zillow/pipelines/blob/3741cc8933b3ec8ae10480c37f28614a967cda3d/sdk/python/kfp/cli/pipeline.py#L217) and [here](https://github.com/zillow/pipelines/blob/3741cc8933b3ec8ae10480c37f28614a967cda3d/sdk/python/kfp/cli/pipeline.py#L255) too, do you mind fixing those quickly too please?

Comment (Bobgy) [reply sdk/python/kfp/cli/run.py:225]: Thanks for letting me know. Happy to fix them as well

Comment (Bobgy) [reply sdk/python/kfp/cli/run.py:225]: Fixed them too!

**Comment by chensun** [comment]
/lgtm
/approve

Thanks @Bobgy! Would you also cherry pick this into `sdk/release-1.8` branch? or @alexlatchford can help?

**Pull request merged**